api/comments/serializers.py

```diff
@@ -0,0 +1,137 @@
+        self.text = text
+
+
+class CommentSerializer(JSONAPISerializer):
+
+    filterable_fields = frozenset([
+        'deleted'
+    ])
+
+    id = IDField(source='_id', read_only=True)
+    type = TypeField()
+    content = AuthorizedCharField(source='return_content')
+
+    user = JSONAPIHyperlinkedRelatedField(view_name='users:user-detail', lookup_field='pk', lookup_url_kwarg='user_id', link_type='related', read_only=True)
+    node = JSONAPIHyperlinkedRelatedField(view_name='nodes:node-detail', lookup_field='pk', lookup_url_kwarg='node_id', link_type='related', read_only=True)
+    target = JSONAPIHyperlinkedGuidRelatedField(link_type='related', meta={'type': 'get_target_type'})
+    replies = JSONAPIHyperlinkedIdentityField(view_name='comments:comment-replies', lookup_field='pk', link_type='self', lookup_url_kwarg='comment_id')
+    reports = JSONAPIHyperlinkedIdentityField(view_name='comments:comment-reports', lookup_field='pk', lookup_url_kwarg='comment_id', link_type='related', read_only=True)
+
+    date_created = ser.DateTimeField(read_only=True)
+    date_modified = ser.DateTimeField(read_only=True)
+    modified = ser.BooleanField(read_only=True, default=False)
+    deleted = ser.BooleanField(read_only=True, source='is_deleted', default=False)
+
+    # LinksField.to_representation adds link to "self"
+    links = LinksField({})
+
+    class Meta:
+        type_ = 'comments'
+
+    def create(self, validated_data):
+        user = validated_data['user']
+        auth = Auth(user)
+        node = validated_data['node']
+
+        validated_data['content'] = validated_data.pop('return_content')
+        if node and node.can_comment(auth):
+            comment = Comment.create(auth=auth, **validated_data)
+        else:
+            raise PermissionDenied("Not authorized to comment on this project.")
+        return comment
+
+    def update(self, comment, validated_data):
+        assert isinstance(comment, Comment), 'comment must be a Comment'
+        auth = Auth(self.context['request'].user)
+        if validated_data:
+            if 'return_content' in validated_data:
+                comment.edit(validated_data['return_content'], auth=auth, save=True)
+            if validated_data.get('is_deleted', None) is True:
+                comment.delete(auth, save=True)
+            elif comment.is_deleted:
+                comment.undelete(auth, save=True)
+        return comment
+
+    def get_target_type(self, obj):
+        object_type = obj._name
+        if not object_type or object_type not in ['comment', 'node']:
+            raise InvalidModelValueError('Invalid comment target.')
+        return object_type
+
+
+class CommentDetailSerializer(CommentSerializer):
+    """
+    Overrides CommentSerializer to make id required.
+    """
+    id = IDField(source='_id', required=True)
+    deleted = ser.BooleanField(source='is_deleted', required=True)
+
+
+class CommentReportsSerializer(JSONAPISerializer):
```

Comment:
I believe the convention is to use the singular, i.e. `CommentReportSerializer`.